src/main/java/com/coreos/jetcd/Client.java

```diff
@@ -1,21 +1,24 @@
 package com.coreos.jetcd;
 
+import com.coreos.jetcd.exception.AuthFailedException;
+import com.coreos.jetcd.exception.ConnectException;
+
 /**
  * Etcd Client.
  */
 public interface Client {
 
-  Auth getAuthClient();
+  Auth getAuthClient() throws AuthFailedException, ConnectException;
```

Comment:
let's do unchecked exceptions then. I think we should have `AuthFailedException`  and `ConnectException` to extend `EtcdException` and use `EtcdExceptionFactory` for creation. 
e.g `EtcdExceptionFactory.newAuthFailedException(string msg, throwable cause)`.